in_toto/in_toto_keygen.py

```diff
@@ -0,0 +1,170 @@
+    as pass phrase.
+    The private key is stored to <'filepath'> and the public key to
+    <'filepath'>.pub
+
+  <Arguments>
+    filepath:
+      The public and private key files are saved to <filepath>.pub, <filepath>
+      respectively
+    password: (optional)
+      If specified the password is used to encrypt the private key
+     bits: (optional)
+      The key size, or key length, of the RSA key. 'bits' must be 2048, or
+      greater, and a multiple of 256.
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError, if the arguments are
+    improperly formatted
+
+  <Side Effects>
+    Writes key files to '<filepath>' and '<filepath>.pub'
+
+  <Returns>
+    None.
+  """
+  securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
+
+  rsa_key = securesystemslib.keys.generate_rsa_key(bits)
+
+  public = rsa_key["keyval"]["public"]
+  private = rsa_key["keyval"]["private"]
+
+  if password:
+    private_pem = securesystemslib.keys.create_rsa_encrypted_pem(
+        private, password)
+  else:
+    private_pem = private
+
+  with open(filepath + ".pub", "w") as fo_public:
+    fo_public.write(public.encode("utf-8"))
+
+  with open(filepath, "w") as fo_private:
+    fo_private.write(private_pem.encode("utf-8"))
+
+def prompt_password(prompt="Enter password: "):
+  """Prompts for password input and returns the password. """
+  return getpass.getpass(prompt, sys.stderr)
+
+def prompt_generate_and_write_rsa_keypair(filepath, bits):
+  """Prompts for password and calls
+  generate_and_write_rsa_keypair"""
+  password = prompt_password()
+  generate_and_write_rsa_keypair(filepath, bits, password)
+
+
+def parse_args():
+  """
+  <Purpose>
+    A function which parses the user supplied arguments.
+
+  <Arguments>
+    None
+
+  <Exceptions>
+    None
+
+  <Returns>
+    Parsed arguments (args object)
+  """
+  parser = argparse.ArgumentParser(
+    description="in-toto-keygen : Generates the keys, stores them with the "
+                "supplied name (public key as: <name>.pub, private key as: "
+                "<name>), additionally prompts for a password when -p is "
+                "supplied and encrypts the private key with the same, "
+                "before storing")
+
+  in_toto_args = parser.add_argument_group("in-toto-keygen options")
+
+  in_toto_args.add_argument("-p", "--prompt", action="store_true",
+                            help="Prompts for a password and encrypts the "
+                            "private key with the same before storing")
+
+  in_toto_args.add_argument("name", type=str,
+                            help="The filename of the resulting key files",
+                            metavar="<filename>")
+
+  in_toto_args.add_argument("bits", nargs= "?", default=3072, type=int,
+                            help="The key size, or key length, of the RSA "
+                            "key.  'bits' must be 2048, or greater, and a "
+                            "multiple of 256.", metavar="<bits>")
```

Comment:
Latter is only [partially true](https://github.com/secure-systems-lab/securesystemslib/blob/master/securesystemslib/keys.py#L211-L214) also see the [default crypto library](https://github.com/secure-systems-lab/securesystemslib/blob/master/securesystemslib/settings.py#L49).
I think you can just drop the info here.